### PR TITLE
Adds ashwalker shamans

### PR DESCRIPTION
**Pull request by Moltijoe**
Adds more diversity to the ashwalker role
They do surgery slightly faster and have a touch heal spell, but take more damage
Only one shaman can be alive at any point in time, an egg only spawns if there is no living shaman and no active shaman egg.

Just need someone to do the flavour stuff for me, @Skrem7 probably

:cl:  @Moltijoe, Tawaporah Koko
rscadd: Adds ashwalker shamans that take more damage, but have a healing touch
tweak: goliath gloves can no longer be cut into fingerless gloves
/:cl:

### CODE REVIEW

**Comment by iamthesilly** [comment]
i love this

**Comment by Moggram** [comment]
cool

**Comment by ChuChu42** [comment]
unga bunga i smash shaman's head with rock

**Comment by MajManatee** [comment]
ah yes, 0.05 more damage

**Comment by Skrem7** [comment]
what do I need to do

**Comment by Moltijoe** [comment]
> what do I need to do

either you or koko should probably give me a proper description of why shamans exist and what sort of power they're using to heal

basically, what should I put here
image
image

**Comment by Moggram** [comment]
>give me a proper description of why shamans exist and what sort of power they're using to heal

I can try to help.

Most shamans in books and RPGs use the magic from everyone in the tribe to cast spells. This is also why there can only be one shaman at any time. As to why the ashwalkers have a shaman is up to the wiki guys.  A good description for the spell would be something like this.

☻Tooltip
"Gather healing magic from tribe."

☻Spell cast
"Soft green light fills your hand."
"Cuts mid, burns soothe, be whole again."

File: code/game/objects/structures/ghost_role_spawners.dm [hunk 1]
```diff
@@ -54,15 +53,26 @@
...
 	var/area/A = get_area(src)
 	team = ashteam
 	if(A)
-		notify_ghosts("An ash walker egg is ready to hatch in \the [A.name].", source = src, action=NOTIFY_ATTACKORBIT, flashwindow = FALSE, ignore_key = POLL_IGNORE_ASHWALKER)
+		notify_ghosts("[mob_name] egg is ready to hatch in \the [A.name].", source = src, action=NOTIFY_ATTACKORBIT, flashwindow = FALSE, ignore_key = POLL_IGNORE_ASHWALKER)
+
+//Ash walker shaman eggs: Spawns in ash walker dens in lavaland. Only one can exist at a time and have the sole purpose of keeping other ashwalkers alive.
+/obj/effect/mob_spawn/human/ash_walker/shaman
+	name = "ash walker shaman egg"
+	desc = "A man-sized yellow egg, spawned from some unfathomable creature. A humanoid silhouette lurks within."
```
Comment (Skrem7) [inline code/game/objects/structures/ghost_role_spawners.dm]:
```suggestion
	desc = "A man-sized, amber egg spawned from some unfathomable creature. A humanoid silhouette lurks within."
```

File: code/modules/clothing/gloves/color.dm [hunk 2]
```diff
@@ -65,6 +65,7 @@
 	desc = "Rudimentary gloves that aid in carrying."
 	icon_state = "goligloves"
 	item_state = "goligloves"
+	can_be_cut = FALSE //no turning goliath gloves into fingerless gloves
```
Comment (Skrem7) [inline code/modules/clothing/gloves/color.dm]: please don't have this it'll cause cornflakes with the BIG bow pr

File: code/game/objects/structures/ghost_role_spawners.dm [hunk 3]
```diff
@@ -54,15 +53,26 @@
...
+	mob_name = "an ash walker shaman"
+	mob_species = /datum/species/lizard/ashwalker/shaman
+	outfit = /datum/outfit/ashwalker/shaman //might be OP, but the flavour is there
+	short_desc = "You are an ash walker shaman. Your tribe worships the Necropolis."
+	flavour_text = "The wastes are sacred ground, its monsters a blessed bounty. \
```
Comment (Skrem7) [inline code/game/objects/structures/ghost_role_spawners.dm]:
```suggestion
	flavour_text = "The wastes are sacred ground, its monsters a blessed bounty. You and your people have become one with the tendril and its land. \
```

File: code/game/objects/structures/ghost_role_spawners.dm [hunk 4]
```diff
@@ -54,15 +53,26 @@
...
+	You have seen lights in the distance... they foreshadow the arrival of outsiders that seek to tear apart the Necropolis and its domain. Fresh sacrifices for your nest."
```
Comment (Skrem7) [inline code/game/objects/structures/ghost_role_spawners.dm]:
```suggestion
	You have seen lights in the distance and from the skies: outsiders that come with greed in their hearts. Fresh sacrifices for your nest."
```

File: code/modules/mob/living/carbon/human/species_types/lizardpeople.dm [hunk 5]
```diff
@@ -161,6 +161,64 @@
...
+
+//removes the heal spell
+/datum/species/lizard/ashwalker/shaman/on_species_loss(mob/living/carbon/C)
+	. = ..()
+	if(goodtouch)
+		C.RemoveSpell(goodtouch)
+
+//basic touch ability that heals brute and burn, only accessed by the ashwalker shaman
+/obj/effect/proc_holder/spell/targeted/touch/healtouch
+	name = "Healing touch"
```
Comment (Skrem7) [inline code/modules/mob/living/carbon/human/species_types/lizardpeople.dm]:
```suggestion
	name = "healing touch"
```

File: code/modules/mob/living/carbon/human/species_types/lizardpeople.dm [hunk 6]
```diff
@@ -161,6 +161,64 @@
...
+	desc = "This spell charges your hand with vile energy that can be used to violently explode victims with healing."
```
Comment (Skrem7) [inline code/modules/mob/living/carbon/human/species_types/lizardpeople.dm]:
```suggestion
	desc = "This spell charges your hand with the vile energy of the Necropolis, permitting you to undo some external injuries from a target."
```

File: code/modules/mob/living/carbon/human/species_types/lizardpeople.dm [hunk 7]
```diff
@@ -161,6 +161,64 @@
...
+	panel = "Ashwalker"
+	charge_max = 20 SECONDS
+	clothes_req = FALSE
+	antimagic_allowed = TRUE
+
+	action_icon_state = "spell_default"
+
+/obj/item/melee/touch_attack/healtouch
+	name = "\improper healing touch"
+	desc = "This hand of mine glows with an awesome power!"
```
Comment (Skrem7) [inline code/modules/mob/living/carbon/human/species_types/lizardpeople.dm]:
```suggestion
	desc = "A blaze of life-granting energy from the hand. Heals minor to moderate injuries."
```

File: code/modules/mob/living/carbon/human/species_types/lizardpeople.dm [hunk 8]
```diff
@@ -161,6 +161,64 @@
...
+	catchphrase = "BE HEALED!!"
```
Comment (Skrem7) [inline code/modules/mob/living/carbon/human/species_types/lizardpeople.dm]:
```suggestion
	catchphrase = "BE REPLENISHED!!"
```

**Review by Skrem7 — CHANGES_REQUESTED** [review]
flavor changes

File: code/modules/clothing/gloves/color.dm (hunk 2 above)
Comment (Moltijoe) [reply code/modules/clothing/gloves/color.dm]: fine, but shamans spawn with the clothing set that has those, and them being cuttable into fingerless gloves seems weird

File: code/modules/mob/living/carbon/human/species_types/lizardpeople.dm (hunk 6 above)
Comment (Moltijoe) [reply code/modules/mob/living/carbon/human/species_types/lizardpeople.dm]: this was literally copy pasted from disintegration desc and then "with healing" tacked on, half expected you to replace the entire line

**Review by JamieD1 — APPROVED** [review]

**Pull request merged**